### PR TITLE
Add API support to the shell

### PR DESCRIPTION
**Pull request by SferaDev**
Add API methods to the shell:

```
api.workspaces.createWorkspace({ name: "foo", slug: "bar" })
```

```
GET /dbs
```

### CODE REVIEW

File: packages/shell/src/cli-run.ts [hunk 1]
```diff
@@ -117,3 +147,15 @@ async function setupHistory(replServer: repl.REPLServer) {
...
+  databaseURL = databaseURL || getDatabaseURL() || '';
+  const [protocol, , host, , database] = databaseURL.split('/');
+  const [workspace] = (host || '').split('.');
+  return {
+    protocol,
+    host,
+    database,
+    workspace
+  };
+}
```
Comment (SferaDev) [inline packages/shell/src/cli-run.ts:161]: This is from the CLI, I think more and more a util package would be appreciated 😄

Comment (gimenete) [reply packages/shell/src/cli-run.ts:162]: Yep. Or we can put it in the `xata/client` one.

File: packages/shell/src/cli-run.ts [hunk 2]
```diff
@@ -40,6 +41,27 @@ export async function run(options: { env?: string; databaseURL?: string; apiKey?
     )} global variable.`
   );
 
+  const fetchApi = async (method: string, path: string, body?: any) => {
+    const { protocol, host } = parseDatabaseURL(databaseURL);
+    // TODO: Add support for staging, how?
```
Comment (gimenete) [inline packages/shell/src/cli-run.ts:46]: What about getting the `host` returned by `parseDatabaseURL`?

**Review by gimenete — APPROVED** [review]
This is great! 😍 

I was thinking on maybe having a different command/REPL, but I think it is not necessary now. It's pretty clear what is an API call and what is a JS expression, and it's cool to have both combined and be able to use both at the same time.

Great job!! 🚀

File: packages/shell/src/cli-run.ts (hunk 2 above)
Comment (SferaDev) [reply packages/shell/src/cli-run.ts:46]: Yeah, that could work. The CLI uses an env variable for changing the API host. Maybe that's a better approach. I'll leave the comment and let's revisit it with usage.

**Pull request merged**